diff --git a/service/stacks/zephyr/sal_adapter_le_interface.c b/service/stacks/zephyr/sal_adapter_le_interface.c
index 9da3487..7179eda 100644
--- a/service/stacks/zephyr/sal_adapter_le_interface.c
+++ b/service/stacks/zephyr/sal_adapter_le_interface.c
@@ -2062,13 +2062,13 @@ bt_status_t bt_sal_le_set_security_level(bt_controller_id_t id, uint8_t level)
 #endif
 }
 
-#ifdef CONFIG_BT_SMP
 static void zblue_convert_le_addr(bt_address_t* addr, ble_addr_type_t type, bt_addr_le_t* le_addr)
 {
     le_addr->type = zblue_convert_addr_type(type);
     memcpy(le_addr->a.val, addr, sizeof(addr->addr));
 }
 
+#ifdef CONFIG_BT_SMP
 static void STACK_CALL(remove_bond)(void* args)
 {
     sal_adapter_req_t* req = args;
