diff --git a/subsys/bluetooth/host/id.c b/subsys/bluetooth/host/id.c
index b1cc2d303bb..a3c64406993 100644
--- a/subsys/bluetooth/host/id.c
+++ b/subsys/bluetooth/host/id.c
@@ -1670,12 +1670,12 @@ int bt_setup_public_id_addr(struct bt_dev *hdev)
 		return 0;
 	}
 
+#if defined(CONFIG_BT_PRIVACY)
 	if (!irk_is_empty(hdev->irk[BT_ID_DEFAULT])) {
 		irk = hdev->irk[BT_ID_DEFAULT];
 		goto out;
 	}
 
-#if defined(CONFIG_BT_PRIVACY)
 	uint8_t ir_irk[16];
 	uint8_t ir[16] = { 0 };
 
@@ -1692,7 +1692,9 @@ int bt_setup_public_id_addr(struct bt_dev *hdev)
 	}
 #endif /* defined(CONFIG_BT_PRIVACY) */
 
+#if defined(CONFIG_BT_PRIVACY)
 out:
+#endif
 	/* If true, `id_create` will randomize the IRK. */
 	if (!irk && IS_ENABLED(CONFIG_BT_PRIVACY)) {
 		/* `id_create` will not store the id when called before BT_DEV_READY.
