diff --git a/boards/sf32lb52/lckfb_huangshan_pi/src/sifli_ap.c b/boards/sf32lb52/lckfb_huangshan_pi/src/sifli_ap.c
index 89ae48e..0897df3 100644
--- a/boards/sf32lb52/lckfb_huangshan_pi/src/sifli_ap.c
+++ b/boards/sf32lb52/lckfb_huangshan_pi/src/sifli_ap.c
@@ -404,6 +404,15 @@ int sf32lb52_lchspi_ulp_bringup(void)
     }
 #endif
 
+#ifdef CONFIG_AUDIO
+  ret = sf32lb_audio_initialize();
+  if (ret < 0)
+    {
+      syslog(LOG_ERR, "ERROR: sf32lb_audio_initialize failed: %d\n", ret);
+      return ret;
+    }
+#endif
+
 #ifdef CONFIG_TIMER
 #if SF32LB_TIMER_DEFAULT_INDEX >= 0
   {

